lib/factory_bot/definition_proxy.rb

```diff
@@ -94,9 +94,10 @@ def method_missing(name, *args, &block) # rubocop:disable Style/MethodMissing
       elsif args.first.respond_to?(:has_key?) && args.first.has_key?(:factory)
         association(name, *args)
       else
-        raise NoMethodError.new(
-          "undefined method '#{name}' in '#{@definition.name}' factory",
-        )
+        raise NoMethodError.new(<<~MSG)
+          undefined method '#{name}' in '#{@definition.name}' factory
+          Did you mean? '#{name} { #{args.first.inspect} }'
```

Comment:
I see that `args.first` is used a few times in this method. Would it be worth extracting a variable to help describe why `args.first` is significant?

What do you think about:

```rb
def method_missing(name, *args, &block)
  attribute_value = Array(args).first  # coercing with `Array()` might not be necessary with splats

  if attribute_value.nil?
    # ...
  elsif attribute_value.respond_to?(:has_key?) && attribute_value.has_key?(:factory)
    # ...
  else
    # ...
  end
end

Reply:
I like it. I adjusted the name a little and pulled out a private predicate method. What do you think:

```rb
def method_missing(name, *args, &block)
  association_options = args.first # AFAIK args will always be an array

  if association_options.nil?
    # ...
  elsif __valid_association_options?__(association_options)
    # ...
  else
    # ...
  end
end

private

def __valid_association_options?(options)
  options.respond_to?(:has_key?) && options.has_key?(:factory)
end
```